README.md
# 2022web_site

#### For Portfolios.
## 포트폴리오 페이지입니다.

1. gabia https://passenger-han.github.io/2022web_site/01site_GabiaGlobal   
2. HiMEDIA https://passenger-han.github.io/2022web_site/02site_himedia
3. Netmarble Foundation https://passenger-han.github.io/2022web_site/03site_NetmarbleFounda
4. Konkuk University https://passenger-han.github.io/2022web_site/04site_college
6. Tour Site - Lotte https://passenger-han.github.io/2022web_site/06site_Tour
7. Neoforet https://passenger-han.github.io/2022web_site/07site_Neoforet_collabo
8. Company https://passenger-han.github.io/2022web_site/08site_Company
9. (Test) todoList https://passenger-han.github.io/2022web_site/09test_todoList
10. Bootstrap5 https://passenger-han.github.io/2022web_site/10site_bootstrap
11. HIRA https://passenger-han.github.io/2022web_site/11site_hira

Changelog.md
----------------
### 20221007 18:20
 11site_Hira
* Have to make scripts for close-button in sitemap at mobile devices..
   
----------------
### 20221006 18:10
 11site_Hira
* Need to refine sitemap for mobile devices..
   
----------------
### 20220930 18:20
 11site_Hira
* should check 'page-4 guideline-visual' script..
   
----------------
### 20220926 18:13
 11site_HIRA
* Sorry, forgot writing the log.
  * I reckon I should learn how to write .markdown efficient and stylish.
   
----------------
### 20220825 18:15
 10site_bootstrap
* 1.0 ..?
   
----------------
### 20220824 18:05
 10site_bootstrap
* ahoufhghegjpoeq
* have to put animations into some items.
   
----------------
### 20220823 18:11
 10site_bootstrap
* qkek qhrh tlvek.. bb....
   
----------------
### 20220822 18:16
 10site_bootstrap
* qkekrnrud rkrh tlvek..
   
----------------
### 20220819 18:14
 10site_bootstrap
* Less than yesterday but I'm still DUCKED LOL
   
----------------
### 20220818 18:20
 10site_bootstrap
* check adaptive page.. I'm DUCKED LOL
   
----------------
### 20220812 18:18
 09test_todoList
* Uploaded for test.
  * 10site_bootstrap
* Using bootstrap..
   
----------------
### 20220805 18:26
 08site_Company
* mainpage ===> news section: colorbox..? O -o;
   
----------------
### 20220804 18:14
 08site_Company
* mainpage ===> news section: a tag wrapper for img tag needed (must NOT work when mobile screen!)
   
----------------
### 20220803 18:20
 08site_Company
* mainpage ===> mobile menu: I think I don't have to fix it! ^ -^
   
----------------
### 20220802 18:20
 08site_Company
* Subpages placeholding..
* Need check mainpage ===> mobile menu: overlay not hidden when toggle menu is on and then window size is wide again.
   
----------------
### 20220729 18:22
 08site_Company
* Do I have to put animating features in?
   
----------------
### 20220728 18:18
 08site_Company
* news-modal hasn't been finished yet..
   
----------------
### 20220726 18:15
 08site_Company
* putting background-images..
   
----------------
### 20220721 18:18
 08site_Company
* Stopped at #sct01. (for making it responsive)
   
----------------
### 20220720 18:18
 08site_Company
* Header.
   
----------------
### 20220719 18:20
 08site_Company
* NOW STARTED.
   
----------------
### 20220701 18:20
 06site_Tour
* mainpage almost!
   
----------------
### 20220628 18:19
 06site_Tour
* progressing, main section almost done..
   
----------------
### 20220627 18:19
 06site_Tour
* swiper added.
   
----------------
### 20220624 16:19
 06site_tour
* form boards are bad things.. I'm mentally almost dying! O<-<
   
----------------
### 20220623 18:20
 06site_tour
* started!
   
----------------
### 20220617 18:21
 04site_college
* 1.4 : modal added
  * 05site_PUBGesports
* Header: learning for flexbox needed..
   
----------------
### 20220616 18:15
 04site_college
* 1.31 : button hover effects added, some images retoned, black boards added for readabilities
  * 05site_PUBGesports started
   
----------------
### 20220615 17:44
 04site_college
* 1.0 released!
* 1.1 : fixed the section 3's p-tags were not in an a-tag.
   
----------------
### 20220613 18:36
 04site_college
* almost (footer left)
   
----------------
### 20220613 18:36
 04site_college
* bxSlider injecting..
   
----------------
### 20220610 18:30
 04site_college
* did header work.
* making visual section..
   
----------------
### 20220607 18:17
 03site_NetmarbleFounda:
* mirror 1.0, maybe T vT
* animate CSS and WOW js added
* O<-<
   
----------------
### 20220602 18:19
 03site_NetmarbleFounda:
* Footer rewrote
* transition added
   
----------------
### 20220531 18:20
 03site_NetmarbleFounda:
* O<-<
   
----------------
### 20220526 18:21
 02site_practice:
* float, position, flex.. too hard to use them all fluently..
* What makes publishers better?
   
----------------
### 20220525 18:23
 02site_practice:
* too slow O<-<
   
----------------
### 20220524 18:20
 01site_selfmaking:
* need to learn how I should resize images..
  * 02site_himedia:
* making header + menu..
   
----------------
### 20220523 18:20
 01site_selfmaking:
* progressing..
   
----------------
### 20220520 18:18
 01site_selfmaking:
* header realigned by using flex mechanisms.
* bxSlider added in the banner.
   
----------------
### 20220519 18:20
 01site_selfmaking: practicing..
* Couldn't find the proper way for logo to get aligned.
* The last thing I took: .banner_visual
   
----------------
### 20220519 16:58
 Synchronizing subfolders and items inside between folder "01site" and folder "01site_selfmaking". Affected items are as below:
* folder: images

11site_hira/js/script.js
// resize check (not used)
let resizeDelay = 300; // ms
let resizeTimer = null;

// $(window).on('resize', function(){
//     clearTimeout(resizeTimer);
//     resizeTimer = setTimeout(function(){
//         if (window.innerWidth >= 960){
//             $('header #header-navigator .navigator').show();
//         } else {
//             $('header #header-navigator .navigator').hide();
//         }
//     }, resizeDelay);
// })

// navigator coloring
$(window).on('scroll', function(){

})

// navigator sitemap check
let sitemapOpened = false;

// navigator full-drop submenu
$('header #header-navigator .navigator').on('mouseenter', function(){
    if (window.innerWidth >= 960 && sitemapOpened == false){
        $('header #header-navigator .navigator .submenu').css({
            'display': 'block',
        });
        $('.background-fulldrop').css({
            'display': 'block',
        });
        $('.background-overlay').css({
            'display': 'block',
        });
    }
});

$('header #header-utility').on('mouseenter', function(){
    $('header #header-navigator .navigator .submenu').css({
        'display': 'none',
    });
    $('.background-fulldrop').css({
        'display': 'none',
    });
    $('.background-overlay').css({
        'display': 'none',
    });
});

$('.background-overlay').on('mouseenter', function(){
    $('header #header-navigator .navigator .submenu').css({
        'display': 'none',
    });
    $('.background-fulldrop').css({
        'display': 'none',
    });
    $('.background-overlay').css({
        'display': 'none',
    });
});

// navigator sitemap
$('header #header-navigator button').on('click', function(){
    $('.sitemap').css({
        'display': 'block',
    });
    $('.background-sitemap').css({
        'display': 'block',
    });

    $('header #header-navigator .navigator .submenu').css({
        'display': 'none',
    });
    $('.background-fulldrop').css({
        'display': 'none',
    });
    $('.background-overlay').css({
        'display': 'none',
    });

    sitemapOpened = true;
});
$('.background-sitemap').on('click', function(){
    $('.sitemap').css({
        'display': 'none',
    });
    $('.background-sitemap').css({
        'display': 'none',
    });

    sitemapOpened = false;
});

// Sitemap at mobile devices
$('.sitemap .sitemap-depth01 > li > a').on('click', function(event){
    event.stopPropagation();
    // 이벤트 발생 대상을 li 로 하면 메뉴 아무거나 눌러도 결국 닫힌다..
    // 근데 closest()를 매개변수 없이는 못써..? 이거 반 쪽짜리야? 그냥 바로 위에꺼 고르면 되는거 아니야?
    
    if ($(this).closest('li').find('.sitemap-depth02-wrapper').hasClass('active')){
        $(this).closest('li').find('.sitemap-depth02-wrapper').removeClass('active');
    } else {
        $('.sitemap-depth02-wrapper').removeClass('active');
        $('.sitemap-depth02').removeClass('active');
        $(this).closest('li').find('.sitemap-depth02-wrapper').addClass('active');
    }
});

$('.sitemap-depth01 > li .sitemap-depth02-wrapper .depth02-title').on('click', function(event){
    event.stopPropagation();
    if ($(this).closest('ul').hasClass('active')){
        $(this).closest('ul').removeClass('active');
    } else {
        $('.sitemap-depth02').removeClass('active');
        $(this).closest('ul').addClass('active');
    }
});

// page-1 latest board
$('#section-01 .panel .categories a').on('click', function(){
    $('#section-01 .panel .categories a').removeClass('on');
    $('#section-01 .panel .categories a').eq($(this).index()).addClass('on');
    loadButtons($('#section-01 .panel .categories a').index(this));
});

function loadButtons(index){
    $('#section-01 .panel .buttons').removeClass('on')
    $('#section-01 .panel .buttons').eq(index).addClass('on')
}

// page-2
$('#section-02 .information li').on('click', function(){
    $('#section-02 .information li').removeClass('on');
    $('#section-02 .information li').eq($(this).index()).addClass('on');
    loadInformation($('#section-02 .information li').index(this));
});

function loadInformation(index){
    $('#section-02 .notice').removeClass('on')
    $('#section-02 .notice').eq(index).addClass('on')
}


// page-3 latest board
$('#section-03 .news-menu a').on('click', function(){
    $('#section-03 .news-menu a').removeClass('on');
    $('#section-03 .news-menu a').eq($(this).index()).addClass('on');
    loadNews($('#section-03 .news-menu a').index(this));
});

function loadNews(index){
    $('#section-03 .news-list').removeClass('on')
    $('#section-03 .news-list').eq(index).addClass('on')
}

$('#section-03 .sns-menu .sns-list li').on('click', function(){
    $('#section-03 .sns-menu .sns-list li').removeClass('on');
    $('#section-03 .sns-menu .sns-list li').eq($(this).index()).addClass('on');
    loadSNS($('#section-03 .sns-menu .sns-list li').index(this));
});

function loadSNS(index){
    $('#section-03 .sns-wrapper .swiper').removeClass('on')
    $('#section-03 .sns-wrapper .swiper').eq(index).addClass('on')
}



// page-4 guideline-visual
$('#section-04 .guideline-list p').on('click', function(){
    $('#section-04 .guideline-list p').removeClass('on');
    // $('#section-04 .guideline-list p').eq($(this).index()).addClass('on');
    $(this).addClass('on');
    loadGuidelineVisual($('#section-04 .guideline-list p').index(this));
});

function loadGuidelineVisual(index){
    $('#section-04 .guideline-visual li').removeClass('on')
    $('#section-04 .guideline-visual li').eq(index).addClass('on')
}

// footer outlink
$('footer .footer-upper .footer-sitemap').on('click', function(){
    $('footer .footer-upper .footer-sitemap ul').toggleClass('on');
})

// page-1 Swiper
const sct01Swiper = new Swiper('#section-01 .swiper', {
    loop: true,
    autoplay: {
        delay: 6000,
        disableOnInteraction: true,
    },
    mousewheel: false,
});

// page-3 Swipers
const sct03Swiper01 = new Swiper ('#section-03 .banner-wrapper .swiper', {
    slidesPerView: 1,
    allowTouchMove: true,
    
    pagination: {
        el: '#section-03 .banner-wrapper .swiper-pagination',
        type: 'fraction',
        clickable: true,
    },
});

const sct03Swiper0201 = new Swiper ('#section-03 .sns-wrapper .swiper01', {
    // loop: true,
    slidesPerView: 2,
    spaceBetween: 16,

    pagination: {
        el: '#section-03 .sns-wrapper .swiper-pagination',
        type: 'fraction',
        clickable: true,
    },

    breakpoints: {
        960: {
            slidesPerView: 4,
            spaceBetween: 16,
        }
    },
});
const sct03Swiper0202 = new Swiper ('#section-03 .sns-wrapper .swiper02', {
    // loop: true,
    slidesPerView: 2,
    spaceBetween: 16,

    pagination: {
        el: '#section-03 .sns-wrapper .swiper-pagination',
        type: 'fraction',
        clickable: true,
    },

    breakpoints: {
        960: {
            slidesPerView: 4,
            spaceBetween: 16,
        }
    },
});
const sct03Swiper0203 = new Swiper ('#section-03 .sns-wrapper .swiper03', {
    // loop: true,
    slidesPerView: 2,
    spaceBetween: 16,

    pagination: {
        el: '#section-03 .sns-wrapper .swiper-pagination',
        type: 'fraction',
        clickable: true,
    },

    breakpoints: {
        960: {
            slidesPerView: 4,
            spaceBetween: 16,
        }
    },
});

// page-4 Swiper
const sct04Swiper01 = new Swiper ('#section-04 .swiper', {
    allowTouchMove: false,
    slidesPerView: 1,
    spaceBetween: 16,
    navigation: {
        prevEl: '#section-04 .navigation-prev',
        nextEl: '#section-04 .navigation-next',
    },
})

// Fullpage.js fullpage initialization
$('#fullpage').fullpage({
    menu: 'header',
    showActiveTooltip: true,
    // anchors: ['page-1', 'page-2', 'page-3', 'page-4'],
    navigation: true,
    normalScrollElements: '#section-04 .guideline-menu .guideline-list, .sitemap',
    responsiveWidth: 960,

    afterResponsive: function(isResponsive){
        if (!isResponsive) {
            $('header #header-navigator .navigator').show();
        } else {
            $('header #header-navigator .navigator').hide();
            $('header #header-navigator .navigator .submenu').hide();
            $('.background-fulldrop').hide();
            $('.background-overlay').hide();
        }
    },

    afterLoad: function(anchorLink, index){
        if (index != 1){
            $('header').addClass('nonTransparent');
            $('header h1 img').attr('src', 'img/logo-filled.png');
        } else {
            $('header').removeClass('nonTransparent');
            $('header h1 img').attr('src', 'img/logo.png');
        }
    }
});

// disable Fullpage.js footer tooltip
$('#fp-nav ul > li:last-child').css('display', 'none');

03site_NetmarbleFounda/js/notice.js
'use strict'

const notices = [
    {
        url: 'https://www.netmarblefoundation.org/notice/view/337',
        title: '제14회 게임콘서트 개최 안내',
        titleImg: 'notice-01-01.jpg',
        desc: '',
        category: '공지사항',
        date: '2022. 09. 23',
    },
    {
        url: 'https://www.netmarblefoundation.org/notice/view/334',
        title: '넷마블게임박물관 건립준비캠페인',
        titleImg: 'notice-01-02.jpg',
        desc: '넷마블문화재단은 게임의 역사 및 문화에 대한 재조명을 통해 한국 게임 산업 발전에 기여할 <넷마블게임박물관> 건립을 준비하고 있습니다. 게임의 과거와 현재 그리고 미래를 그려나갈 넷마블게임박물관에 게임 유물 기증과 함께 여러분의 의견을 들려주세요!',
        category: '공지사항',
        date: '2022. 09. 14',
    },
    {
        url: 'https://www.netmarblefoundation.org/notice/view/334',
        title: '[게임소통학교] 도미노 영상 이벤트 결과 발표',
        titleImg: 'notice-01-03.jpg',
        desc: '',
        category: '공지사항',
        date: '2022. 08. 18',
    },
    {
        url: 'https://www.netmarblefoundation.org/notice/view/317',
        title: '[마감] 넷마블문화재단 사업기획 및 운영 담당자 채용 공고',
        titleImg: '',
        desc: '넷마블문화재단에서 아래와 같이 새로운 직원을 모집합니다. 많은 분들의 지원 바랍니다.',
        category: '공지사항',
        date: '2022. 08. 02',
    },
    {
        url: 'https://www.netmarblefoundation.org/notice/view/314',
        title: '제13회 게임콘서트 개최 안내',
        titleImg: 'notice-01-05.jpg',
        desc: '',
        category: '공지사항',
        date: '2022. 06. 23',
    },
    {
        url: 'https://www.netmarblefoundation.org/notice/view/314',
        title: '2022 넷마블 온라인 게임소통학교 참가 가족 모집(구. 게임소통교육)',
        titleImg: 'notice-01-06.jpg',
        desc: '안녕하세요. 게임에 대한 부모와 자녀 간의 견해 차이를 좁히고, 교육 및 놀이의 도구로서 게임이 가지고 있는 순기능을 전파하고자 진행되는 2022 게임소통학교 신청 관련 안내 드립니다.',
        category: '공지사항',
        date: '2022. 06. 15',
    },
]

export {notices};